packages/core/src/tasks/check-module-boundaries.ts

```diff
@@ -119,6 +120,44 @@ function findProjectGivenRoot(
   }
 }
 
+const regexMap = new Map<string, RegExp>();
+
+function hasMatch(input: string[], pattern: string): boolean {
+  if (pattern === '*') return true;
+
+  // if the pattern is a regex, check if any of the input strings match the regex
+  if (pattern.startsWith('/') && pattern.endsWith('/')) {
+    let regex: RegExp;
+    if (regexMap.has(pattern)) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      regex = regexMap.get(pattern)!;
+    } else {
+      regex = new RegExp(pattern.substring(1, pattern.length - 1));
+      regexMap.set(pattern, regex);
+    }
+    return (input || []).some((t) => regex.test(t));
+  }
+
+  // if the pattern is a glob, check if any of the input strings match the glob prefix
+  if (pattern.includes('*')) {
+    const regex = mapGlobToRegExp(pattern);
+    return (input || []).some((t) => regex.test(t));
+  }
+
+  return (input || []).indexOf(pattern) > -1;
```

Comment:
input is a required parameter, it shouldn't be null so you shouldn't need the `||` handling on it here or above on ln 144